backend/src/main/java/com/redhat/cloud/notifications/db/BehaviorGroupResources.java

```diff
@@ -49,345 +49,328 @@
-                            deleteQuery += " AND endpoint.id NOT IN (:endpointIds)";
-                        }
-                        Mutiny.Query mutinyQuery = session.createQuery(deleteQuery)
-                                .setParameter("behaviorGroupId", behaviorGroupId);
-                        if (!endpointIds.isEmpty()) {
-                            mutinyQuery = mutinyQuery.setParameter("endpointIds", endpointIds);
-                        }
-                        return mutinyQuery.executeUpdate();
-
-                    })
-                    .onItem().transformToMulti(ignored -> Multi.createFrom().iterable(endpointIds))
-                    .onItem().transformToUniAndConcatenate(endpointId -> {
-
-                        /*
-                         * Then, we'll execute an "upsert" based on the given endpointIds list:
-                         * - if an action already exists, its position will be updated
-                         * - otherwise, the action will be inserted into the database
-                         * In the end, all inserted or updated actions will have the same position than the endpointIds list order.
-                         */
-                        String upsertQuery = "INSERT INTO behavior_group_action (behavior_group_id, endpoint_id, position, created) " +
-                                "SELECT :behaviorGroupId, :endpointId, :position, :created " +
-                                "WHERE EXISTS (SELECT 1 FROM endpoints WHERE account_id " +
-                                (accountId == null ? "IS NULL" : "= :accountId") +
-                                " AND id = :endpointId) " +
-                                "ON CONFLICT (behavior_group_id, endpoint_id) DO UPDATE SET position = :position";
-
-                        var sessionQuery = session.createNativeQuery(upsertQuery)
-                                .setParameter("behaviorGroupId", behaviorGroupId)
-                                .setParameter("endpointId", endpointId)
-                                .setParameter("position", endpointIds.indexOf(endpointId))
-                                .setParameter("created", LocalDateTime.now(UTC));
-
-                        if (accountId != null) {
-                            sessionQuery = sessionQuery.setParameter("accountId", accountId);
-                        }
-
-                        return sessionQuery.executeUpdate();
-                    })
-                    .collect().asList()
-                    .replaceWith(OK)
-                    // The following exception will be thrown if the behavior group is not found with the first query.
-                    .onFailure(NoResultException.class).recoverWithItem(NOT_FOUND)
-                    // The following exception will be thrown if the behavior group contains more than one EMAIL_SUBSCRIPTION action.
-                    .onFailure(IllegalArgumentException.class).recoverWithItem(BAD_REQUEST);
```

Comment:
Is this still true? (i.e. Bad request if group contains more than one EMAIL_SUBSCRIPTION

Reply:
I asked myself the same question and found the answer in the git history. The `IllegalArgumentException` was thrown in an older version but then we removed it. That `.onFailure(IllegalArgumentException.class)` is actually useless in the current code.